src/AstToCFlat.ml

```diff
@@ -899,32 +899,57 @@ and mk_expr (env: env) (locals: locals) (e: expr): locals * CF.expr =
 
 (* See digression for [dup32] in CFlatToWasm *)
 let scratch_locals =
-  [ I64; I64; I32; I32 ]
```

Comment:
the top-of-the-stack pointer is now always saved in the fifth local

Reply:
(as opposed to being left on the stack, which used to require contortions to swap operands at the top of the stack)